### DISCUSSION TITLE
Improve UX feedback of password selection

### DISCUSSION THREAD

**Discussion by fubuloubu** [fief-dev/fief#388] [Ideas]
Currently, I think if the password fails complexity requirements (it's hard to tell because the console or logs don't say anything either) you get back a very unhelpful "Invalid email or password" message

It would be good to expand those messages or add a password complexity meter (e.g. poor, moderate, good, great bar) via js or htmx server-side check so that it's clear what the issue is preventing from signing up

I ran into this issue having someone else try to log in who doesn't use randomized passwords, so this was my best guess at what the issue was

**Comment by fubuloubu** [comment]
whoops, misunderstanding on our part, they were trying to use the Sign In page not Sign Up :facepalm:

**Comment by frankie567** [comment]
Ha ha 🙂 Worth to note that there is a password meter (on Sign Up, of course!):

<img width="492" alt="Capture d’écran 2024-05-30 à 16 51 48" src="https://github.com/fief-dev/fief/assets/1144727/56c6209b-185a-41ca-8107-bafcf1dc2ecf">
